tmt/utils.py

```diff
@@ -2459,43 +2459,45 @@ def web_git_url(url: str, ref: str, path: Optional[Path] = None) -> str:
 
 
 @lru_cache(maxsize=None)
-def fmf_id(name: str, fmf_root: Path, always_get_ref: bool = False) -> 'tmt.base.FmfId':
+def fmf_id(name: str, fmf_root: Path, logger: tmt.log.Logger,
+           always_get_ref: bool = False) -> 'tmt.base.FmfId':
     """ Return full fmf identifier of the node """
 
-    def run(command: str) -> str:
+    def run(command: Command) -> str:
         """ Run command, return output """
-        cwd = fmf_root
-        result = subprocess.run(
-            command.split(),
-            stdout=subprocess.PIPE,
-            stderr=subprocess.DEVNULL,
-            cwd=cwd)
-        return result.stdout.strip().decode("utf-8")
+        try:
+            result = run_command(command=command, cwd=fmf_root, logger=logger)
+            if result.stdout is None:
+                return ""
+            return result.stdout.strip()
+        except BaseException:
```

Comment:
`BaseException` is almost always not the right choice, `Exception` or `RunError `would be better.

But, is this the right reaction to all exceptions a command can trigger? I suppose there are commands that may fail,

Reply:
Fixed in 27e8add5.